deps/rabbit/src/rabbit_db.erl

```diff
@@ -0,0 +1,47 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%% Copyright (c) 2007-2022 VMware, Inc. or its affiliates.  All rights reserved.
+%%
+
+-module(rabbit_db).
+
+-include_lib("kernel/include/logger.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+-export([dir/0,
+         run/1]).
+
+%% -------------------------------------------------------------------
+%% dir().
+%% -------------------------------------------------------------------
+
+-spec dir() -> Dir when
+      Dir :: any().
+%% @doc Returns the path where the database files are stored on disk.
+%%
+%% @returns the path to the database directory.
+
+dir() ->
+    rabbit_mnesia:dir().
```

Comment:
This value can't change at runtime. It is mainly used by non-Mnesia-related subsystems, like all Ra systems.

I believe this is a historic bad behavior of RabbitMQ. Instead of defining a place where all subsystems, including Mnesia, write there data (like one directory each), they all write files and directories inside the Mnesia directory. The `rabbit_mnesia:is_virgin_node()` function needs to know every files and directories which can be direct children of the Mnesia directory to distinguish Mnesia data from RabbitMQ data.

Ideally we should move away from this practice. There is a semantic issue there: Khepri relies on a Ra system which relies on the Mnesia directory. Mnesia doesn't have to run because RabbitMQ will create this directory. Perhaps we can get away with renaming some functions internally while keeping the same layout on disk and support for the `$RABBITMQ_MNESIA_DIR` and related environment variables. I don't know, I didn't really think about this yet.

Reply:
This will be addressed by #6462. This patch will be rebased and modified after #6462 is merged.